### ISSUE TITLE
In error: "AssertionError: LivPrb is not a list or time varying distribution"

### ISSUE THREAD

**Issue by sbenthall** [econ-ark/HARK#1305]
```
parameters = {'cycles': 0,
 'aNrmInitStd': 0.0,
 'aNrmInitMean': 6,
 'LivPrb': 0.9949620563926881,
 'PermGroFac': 1.0,
 'pLvlInitMean': 0.0,
 'pLvlInitStd': 0.0,
 'Rfree': 1.0,
 'TranShkStd': [0],
 'PermShkStd': [0],
 'CRRA': 3,
 'DiscFac': 0.99,
 'ex_post': None,
 'AgentCount': 1,
 'num_per_type': 1000,
 'RiskyAvg': 1.0916031021270372,
 'RiskyStd': 0.09296520824951145,
 'T_sim': 3000,
 'UnempPrb': 1.0,
 'IncUnemp': 0.0}

a1 = SequentialPortfolioConsumerType(**parameters)

a1.solve()
```

Gets me this error:

```
AssertionError: LivPrb is not a list or time varying distribution, but should be because it is in time_vary
```

This doesn't seem right.

**Comment by mnwhite** [comment]
This is correct. LivPrb is survival probability, and so it should be a
list. It depends on age, if we're doing a lifecycle model.

**Comment by sbenthall** [comment]
Ok, glad this is fixed with the new parameters object.

@mnwhite I don't follow the logic there. It's not a lifecycle model in this
case.

**Comment by alanlujan91** [comment]
It's not a lifecycle model, but LivPrb is a "lifecycle" attribute, so, it needs to be generated as LivPrb = [ 0.9949620563926881 ] because it's an infinitely repeated "lifecycle" of length one.

**Comment by sbenthall** [comment]
Ok. I'm following now.

It sounds like this is something that's hard-coded in the model which probably ideally is not hard-coded in the model, since we run into cases all the time where we want something that wasn't originally thought of as time-varying to be time-varying, etc.

I suppose https://github.com/econ-ark/HARK/pull/1240 is the fix for this.

Looking forward to that PR!

**Comment by mnwhite** [comment]
Right. For something like ConsIndShock or the portfolio model, *most*
parameters can be "harmlessly" moved between time_vary and time_inv on the
solution side. Because the simulation method is hard-coded with the base
assumptions about what's time-varying, additional work needs to be done on
that side. But with the simulation overhaul to be based on the "model
equations representation", whether a particular object has a t-subscript is
something that can/should be handled at runtime.

In ConsIndShock, the big exception is CRRA. As written, the solution code
assumes that CRRA is constant across periods, and uses this fact in
approximating the upper extrapolation. This should be fixed along side the
change to the timing of expectations, so that it *checks* whether next
period's (marginal) value function identifies itself as coming from the
standard consumption-saving problem. Or we just have a flag that can tell
it not to attempt clever extrapolation.
